statistics/feedback.go

```diff
@@ -501,7 +512,8 @@ func encodePKFeedback(q *QueryFeedback) (*queryFeedback, error) {
 			return nil, errors.Trace(err)
 		}
 		// There is no need to update the point queries.
-		if fb.upper.GetInt64()-fb.lower.GetInt64() <= 1 {
+		result := fb.upper.GetInt64() - fb.lower.GetInt64()
+		if result <= 1 && !(fb.lower.GetInt64() < 0 && result < fb.upper.GetInt64()) {
```

Comment:
What's this check for? And what if the `fb.lower` and `fb.upper` are not of type Int64?